spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/KubernetesConfigDataLocationResolver.java

```diff
@@ -170,25 +169,25 @@ private static PropertyHolder of(ConfigDataLocationResolverContext context) {
 			String namespace = binder.bind("spring.cloud.kubernetes.client.namespace", String.class)
 					.orElse(binder.bind("kubernetes.namespace", String.class).orElse(""));
 
-			KubernetesClientProperties kubernetesClientProperties = clientProperties(context);
```

Comment:
this is a little tricky. What we used to do, is create a `KubernetesClientProperties` and then change its namespace. I do the same thing here, but via a copy constructor in `KubernetesClientProperties` called `withNamespace`, because that is now a record.